components/scream/cime_config/namelist_defaults_scream.xml

```diff
@@ -0,0 +1,154 @@
+<?xml version="1.0"?>
+
+<namelist_defaults>
+
+  <!--
+      The following block defines convenience selectors. Case settings
+      can be used as selectors, but they are often not covenient due to
+      long names or having to process their values through regular expressions
+      to get the subpiece you are interested in. The convenience selectors
+      allow you to express this logic and capture it with a short name.
+
+      The follow attributes are how one defines a selector
+        name     : The name of the selector
+        case_env : The case value (thing you can xmlquery)
+        regex    : OPTIONAL, how to pick out the piece of the value you are interested in
+                   By default, the entire value is grabbed. If the regex does not match
+                   the case value, then the selector will evaluate to False.
+
+      Example using the nlev selector
+        <partmethod nlev="128">42</partmethod>
+        <partmethod nlev="72">420</partmethod>
+  -->
+  <selectors>
+    <selector name="hgrid" case_env="ATM_GRID"/>
+    <selector name="dyn" case_env="ATM_DYN_TARGET"/>
+    <selector name="nlev" case_env="SCREAM_CMAKE_OPTIONS" regex=".*SCREAM_NUM_VERTICAL_LEV ([0-9]+).*"/>
+  </selectors>
+
+  <file name="data/scream_input.yaml" format="yaml">
+    <Atmosphere__Driver>
```

Comment:
I will work to generate a more "plain" tree that can be more easily parsed by buildnml, and easier to display in the defaults file. I think `buildnml` can then put the blocks back together, and generate the current YAML structure again. Tbc, I'm thinking of

```
<atm_procs_names>homme, p3shoc, spa, rad<atm_procs_names>
<homme>
  <Enable__Output__Fields__Checks>False</Enable__Output__Fields__Checks>
  <Grid>dynamics</Grid>
  <!-- other params for homme -->
</homme>
<p3shoc>
  <Process__Type>Gruop</Process__Type>
  <Schedule__Type>Sequential</Schedule__Type>
  <Processes__Names>p3, shoc</Processes__Names>
  ...
</p3shoc>
<p3>
...
</p3>
<spa>
 <SPA__Data__File hgrid="ne4">spa_ne4_file.nc</SPA__Data__File>
 <SPA__Data__File hgrid="ne30">spa_ne30_file.nc</SPA__Data__File>
</spa>
...
<Grids__Manager>
 ...
</Grids__Manager>
<Initial__Conditions>
  <Start__Date> ... </Start_Date>
  <Input__Files>blah, blah2, ... </Input_Files>
</Initial__Conditions>
<Scorpio__Output>
  <Output__YAML__Files> a.yaml, b.yaml </Output__YAML__Files>
</Scorpio__Output>
<Model__Restart>
  <Output__Control>
     <Frequency>10</Frequency>
     <Frequency__Units>Steps</Frequency__Units>
  </Output__Control>
  <Casename>my_restart</Casename>
</Model__Restart>
...
```
This way, our `atmchange` python util will allow for one (maybe two?) indirection layers, like
```
./atmchange Initial__Conditions::Start__Date="2000-01-01.00:00:00"
./atmchange Model__Restart::Output__Control::Frequency=10
...
```
Then, buildnml can put together a yaml file, since we know which sections need to be present, and how they are nested. Better yet, I could flatten a bit our YAML file, without all those layers of nesting.

Note: I think what you did was the best choice, since we don't want this to be a single PR. Adding the defaults reflecting the current input files structure is the best first step. Next, comes structuring the xml files in a more legible/skinny way. Just making a note here.

Reply:
Yes, I think your refactors will simplify the situation. Also, we won't need these annoying double underscores once we switch to "snake case" or whatever you call: this_naming_convention.

Reply:
Yes, I think that's snake_case.